app/views/CreateChannelView.tsx

```diff
@@ -73,357 +74,295 @@ interface IOtherUser {
 	fname: string;
 }
 
-interface ICreateChannelViewState {
-	channelName: string;
-	type: boolean;
-	readOnly: boolean;
-	encrypted: boolean;
-	broadcast: boolean;
-	isTeam: boolean;
-	permissions: boolean[];
-}
-
-interface ICreateChannelViewProps extends IBaseScreen<ChatsStackParamList, 'CreateChannelView'> {
-	baseUrl: string;
-	error: object;
-	failure: boolean;
-	isFetching: boolean;
-	encryptionEnabled: boolean;
-	users: IOtherUser[];
-	user: IUser;
-	teamId: string;
-	createPublicChannelPermission: string[] | undefined;
-	createPrivateChannelPermission: string[] | undefined;
-}
-
 interface ISwitch extends SwitchProps {
 	id: string;
 	label: string;
+	hint: string;
 }
 
-class CreateChannelView extends React.Component<ICreateChannelViewProps, ICreateChannelViewState> {
-	private teamId?: string;
-
-	constructor(props: ICreateChannelViewProps) {
-		super(props);
-		const { route } = this.props;
-		const isTeam = route?.params?.isTeam || false;
-		this.teamId = route?.params?.teamId;
-		this.state = {
-			channelName: '',
-			type: true,
-			readOnly: false,
-			encrypted: false,
-			broadcast: false,
-			isTeam,
-			permissions: []
-		};
-		this.setHeader();
-	}
+const RenderItem = memo(
```

Comment:
`memo` can't memoize objects.
Even though you have a shallow equal there.
Just don't do it.

```ts
interface IOtherUser {
	_id: string;
	name: string;
	fname: string;
}
```